flipt-engine-ffi/src/http.rs

```diff
@@ -322,59 +322,71 @@ impl HTTPFetcher {
+                    let reader = StreamReader::new(
+                        resp.bytes_stream()
+                            .map_err(|err| std::io::Error::new(std::io::ErrorKind::Other, err)),
+                    );
+                    let codec = tokio_util::codec::LinesCodec::new();
+                    let frame_reader = tokio_util::codec::FramedRead::new(reader, codec);
+
+                    let mut stream = frame_reader
+                        .into_stream()
+                        .map(|frame| match frame {
+                            Ok(frame) => match serde_json::from_str::<StreamChunk>(&frame) {
+                                Ok(result) => Ok(result),
+                                Err(e) => Err(Error::InvalidJSON(format!(
+                                    "failed to parse response body: {}",
+                                    e
+                                ))),
+                            },
+                            Err(e) => {
+                                Err(Error::Server(format!("failed to read stream chunk: {}", e)))
+                            }
+                        })
+                        .map(|result| match result {
+                            Ok(result) => match result.result.namespaces.into_iter().next() {
+                                Some((_, doc)) => Ok(doc),
+                                None => {
+                                    let err = Error::Server("no data received from server".into());
+                                    Err(err)
+                                }
+                            },
+                            Err(err) => Err(err),
+                        });
+
+                    while let Some(value) = stream.next().await {
+                        match sender.send(value).await {
+                            Ok(_) => continue,
+                            Err(e) => {
+                                return Err(Error::Server(format!(
+                                    "failed to send result to engine {}",
+                                    e
+                                )))
                             }
-                        },
-                        Err(err) => Err(err),
-                    });
-
-                while let Some(value) = stream.next().await {
-                    match sender.send(value).await {
-                        Ok(_) => continue,
-                        Err(e) => {
-                            return Err(Error::Server(format!(
-                                "failed to send result to engine {}",
-                                e
-                            )))
                         }
                     }
-                }
 
-                Ok(())
+                    // If we've reached here, streaming completed successfully
+                    return Ok(());
+                }
+                Ok(None) => return Ok(()),
+                Err(e) => {
+                    retry_count += 1;
```

Comment:
Nice: should we set the retry count and backoff back to initial values in the positive case?

To avoid depleting these when network errors are sporadic and short term.

Reply:
yeah good call